src/main/java/org/wso2/carbon/module/swiftiso20022/utils/JsonToMt940Utils.java

```diff
@@ -0,0 +1,326 @@
+            return new ErrorModel(ConnectorConstants.ERROR_T51, ConnectorConstants.ERROR_BAL_IND_INVALID);
+        }
+
+        if (balanceDetails.getStatementType() != null && !balanceDetails.getStatementType().isBlank()) {
+            if (!(ConnectorConstants.CURRENT_STATEMENT_TYPE.equals(balanceDetails.getStatementType()) ||
+                    ConnectorConstants.LAST_STATEMENT_TYPE.equals(balanceDetails.getStatementType()))) {
+                return new ErrorModel(ConnectorConstants.ERROR_T51, ConnectorConstants.ERROR_INVALID_STATEMENT_TYPE);
+            }
+        }
+
+        return new ErrorModel();
+    }
+
+    /**
+     * Method to validate the date format.
+     *
+     * @param dateTime  Date time to validate
+     * @return          True if the date format is valid, else false
+     */
+    public static boolean isValidDateFormat(String dateTime) {
+
+        if (dateTime == null || dateTime.isBlank()) {
+            return false;
+        }
+        DateFormat formatter = new SimpleDateFormat(ConnectorConstants.DATE_TIME_FORMAT);
+
+        try {
+            formatter.parse(dateTime);
+            return true;
+        } catch (ParseException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Method to validate the transaction type.
+     *
+     * @param transactionType  Transaction type to validate
+     * @return                 True if the transaction type is valid, else false
+     */
+    public static ErrorModel isValidTransactionType(String transactionType) {
+        ErrorModel errorModel = new ErrorModel();
+        if (!transactionType.startsWith("S") && !transactionType.startsWith("N") &&
+                !transactionType.startsWith("F")) {
+            return new ErrorModel(ConnectorConstants.ERROR_T53,
+                    ConnectorConstants.ERROR_TRANS_TYPE_INVALID);
+        }
+
+        if (transactionType.startsWith("S")) {
+            String identificationCode = transactionType.substring(1);
+            if (!ValidatorUtils.isNumber(identificationCode)) {
+                return new ErrorModel(ConnectorConstants.ERROR_T53,
+                        ConnectorConstants.ERROR_TRANS_TYPE_INVALID);
+            }
+
+            if (!(Integer.parseInt(identificationCode) > 99 && Integer.parseInt(identificationCode) < 1000)) {
+                return new ErrorModel(ConnectorConstants.ERROR_T18,
+                        ConnectorConstants.ERROR_TRANS_TYPE_INVALID);
+            }
+        }
+
+        errorModel.setIsError(false);
+        return errorModel;
+    }
+
+    /**
+     * Method to validate the transaction type.
+     *
+     * @param debitCreditMark  Debit/Credit Mark to validate
+     * @return                 True if the debitCreditMark is valid, else false
+     */
+    public static boolean isValidDebitCreditMark(String debitCreditMark) {
+        return debitCreditMark.startsWith("D") || debitCreditMark.startsWith("C") ||
```

Comment:
Can't we use existing CREDIT, DEBIT constants here?

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/2/commits/8d6b87e8988955fbe9679e7afa9d4285b0ff0ac8